lib/ruby/stdlib/date/format.rb

```diff
@@ -376,9 +376,7 @@ def self._strptime_i(str, fmt, e) # :nodoc:
   private_class_method :_strptime_i
 
   def self._strptime(str, fmt='%F')
-    parser = org.jruby.util.RubyDateParser.new
-    map = parser.parse(JRuby.runtime.current_context, fmt, str)
-    return map.nil? ? nil : map.to_hash.inject({}){|hash,(k,v)| hash[k.to_sym] = v; hash}
+    return JRuby.runtime.current_context.getRubyDateParser.parseAndCreateHash(JRuby.runtime.current_context, fmt, str)
```

Comment:
@muga @kares I am implementing _strptime as native ext since I think I can do this a faster (in fact I think I am already done).  This will invalidate this PR but I will point out which commit it is in.